electron/legendary/games.ts

```diff
@@ -314,10 +315,50 @@ class LegendaryGame extends Game {
       autoInstallDxvk
     } = await this.getSettings()
 
+    const DiscordRPC = makeClient('852942976564723722')
+
+    const { discordRPC } = (await GlobalConfig.get().getSettings())
+    if (discordRPC) {
+      // Show DiscordRPC
+      // This seems to run when a game is updated, even though the game doesn't start after updating.
+      const gameInfo = await this.getGameInfo()
+      let os: string
+
+      switch (process.platform) {
+      case 'linux':
+        os = 'Linux'
+        break
+      case 'win32':
+        os = 'Windows'
+        break
+      case 'darwin':
+        os = 'MacOS'
+        break
+      default:
+        os = 'Unknown OS'
+        break
+      }
+
+      DiscordRPC.updatePresence({
+        details: gameInfo.title,
+        instance: true,
+        largeImageKey: 'icon',
+        large_text: gameInfo.title,
+        startTimestamp: Date.now(),
+        state: 'via Heroic on ' + os
+      })
+    }
+
     if (isWindows) {
       const command = `${legendaryBin} launch ${this.appName} ${launcherArgs}`
       console.log('\n Launch Command:', command)
-      return await execAsync(command)
+      const v = await execAsync(command)
```

Comment:
In this case this wont work because you didn't launch the game here, you're launching it on line 361.
The right order would be to remove this v variable, just keep `await execAsync(command)`, then, at the end, instead of `return v` , your should do return `DicordRPC.disconnect()`.
Since this is the last thing you want to run on the function.